src/atomate2/vasp/sets/base.py

```diff
@@ -1125,7 +1125,7 @@ def _set_kspacing(
                 incar["ISMEAR"] = user_incar_settings.get("ISMEAR", 2)
             else:
                 incar["SIGMA"] = user_incar_settings.get("SIGMA", 0.05)
-                incar["ISMEAR"] = user_incar_settings.get("ISMEAR", -5)
+                incar["ISMEAR"] = user_incar_settings.get("ISMEAR", 0)
```

Comment:
I will let @esoteric-ephemera comment on the ISMEAR setting, but independent of that comment, if we do stick with `ISMEAR = -5`, the `SIGMA` should be removed.

Reply:
Oh that's subtle!

Reply:
> I think you have this backwards. ISMEAR = -5 is very important for narrow-gapped systems. If we know the system has a gap, then we should always use ISMEAR = -5.

That's true for energies but not for forces. Looking at about 370 structures from ICSD that aren't in MP, the deviations between the forces using ISMEAR = -5 and 0 persists for even narrow-gapped materials, see below.

I guess my question is if the `_set_kspacing` function is supposed to be MP-specific or agnostic. If it's agnostic, then I'll just update ISMEAR / SIGMA for the MP flows and revert this back to -5.

![PBE_ISMEAR_forces_bandgap](https://github.com/materialsproject/atomate2/assets/33381112/a4d83647-fb13-403d-aa81-515fd719622a)

Reply:
I'm not sure what you're plotting here? How are you evaluating whether the forces are better with ISMEAR = -5 or 0?

Reply:
Plotting consistency of the smearing methods since we don't have a reference calculation to compare with. The vertical axis is $\sum_{ij} | F_j(R_i,\mathrm{ISMEAR=-5}) -  F_j(R_i,\mathrm{ISMEAR=0})|$, where $R_i$ are the ionic positions, and $j=x, y, z$. Each structure then gets one data point.

You're right that I can't say that one is better except for true metals, where we know that the forces from the tetrahedron method aren't variational. I'm intuiting from this that there are similar errors for very narrow gap insulators and semi-metals.

For systems with very small gaps, the partial occupancies from ISMEAR = 0 are problematic in that they're unphysical, but they should help in accelerating SCF convergence. You can see for the plot below (same set of structures) that the bandgap for very small gap materials is sometimes *larger* with Gaussian smearing than tetrahedron smearing. Sometimes it's less as expected (probably from the partial occupancies).

![PBE_ISMEAR_bandgap_comp](https://github.com/materialsproject/atomate2/assets/33381112/8ca12261-92ad-4b29-9a29-c66f8ab01a27)

Reply:
Ok, thanks for clarifying. These plots are nice to see. Just to respond to your points:

> You're right that I can't say that one is better except for true metals, where we know that the forces from the tetrahedron method aren't variational. I'm intuiting from this that there are similar errors for very narrow gap insulators and semi-metals.

Semi-metals have a band that crosses the Fermi level, and hence you should use ISMEAR = 0 as you're expecting partial occupancies. However, very narrow band gap insulators don't have a band that crosses the Fermi level, hence no partial occupancies. Here using ISMEAR = -5 will give you the most accurate results.

> For systems with very small gaps, the partial occupancies from ISMEAR = 0 are problematic in that they're unphysical, but they should help in accelerating SCF convergence. 

I'm not sure I follow that they will help with accelerating SCF convergence. I believe the main thing ISMEAR = -5 helps with is k-point convergence, and avoiding spurious partial occupancies.

Just as a final note, if you have a small gap material and want to calculate dielectric properties, then ISMEAR = -5 is essential, otherwise VASP will report a metal and the calculation will fail.

So I think its best to revert this change and set ISMEAR = -5.

Reply:
SCF convergence was the wrong phrase - I meant the reduction in needed k-point density. Both methods should help with that for semi-/metals.

I'll change this back. There's a larger ongoing discussion in MP about whether to abandon ISMEAR = -5 for relaxations in insulators (the current default) in favor of ISMEAR = 0. That will require running "fully-converged" calculations for this same set.

Reply:
> There's a larger ongoing discussion in MP about whether to abandon ISMEAR = -5 for relaxations in insulators (the current default) in favor of ISMEAR = 0.

Ok, I would strongly advise against doing that! There are basically no downsides to ISMEAR = -5 to insulators and lots of downsides to ISMEAR = 0.

> That will require running "fully-converged" calculations for this same set.

What do you mean by "fully-converged"

Reply:
Fully converged as in using a very high density of k-points, very high cutoffs, and no smearing (Gaussian + very small smearing).

Wrt downsides: I can see issues where for very small gaps, you get erroneous partial occupancies near the fermi level. Practically, the gap has to be <~ 0.15 eV for the Gaussian smearing we use (SIGMA = 0.05 by default) for virtual states to be significantly partially occupied. Are there other downsides you're thinking of?

Reply:
Thanks for sharing that paper. Looks interesting. 

> Are there other downsides you're thinking of?

To be honest, partial occupancies is the main one. The nice thing is that if you use ISMEAR = -5 for all gapped materials then at least your calculations are then consistent between relaxation, static, and DOS.

Can I ask why you're thinking of moving away from ISMEAR = -5?

Reply:
Two major reasons: since we already have a big transition in going from atomate --> atomate2, we're auditing the parameters MP uses in its WFs (esp. GGA); and there are a few collaborations on dataset generation where this ISMEAR benchmark was requested